lib/logstash/filters/elasticsearch/client.rb

```diff
@@ -22,6 +22,9 @@ def initialize(logger, hosts, options = {})
         transport_options[:headers].merge!(setup_api_key(api_key))
         transport_options[:headers].merge!({ 'user-agent' => "#{user_agent}" })
 
+        transport_options[:pool_max] = 1000
+        transport_options[:pool_max_per_route] = 100
```

Comment:
@edmocosta to avoid adding config options in a patch release, I have added these with the same default values that we have in the referenced Elasticsearch Output plugin, which should allow us to completely eliminate connection contention on pipelines with fewer than 100 workers and 10 `hosts`. We can make this configurable if/when necessary.